File: javaboxuegu/src/main/java/com/xushengling/javaboxuegu/activity/LoginActivity.java
package com.xushengling.javaboxuegu.activity;

import androidx.annotation.Nullable;

import android.content.Intent;
import android.content.SharedPreferences;
import android.text.TextUtils;
import android.widget.EditText;
import android.widget.TextView;


import com.xushengling.javaboxuegu.base.BaseActivity;
import com.xushengling.javaboxuegu.utils.MD5Utils;
import com.xushengling.javaboxuegu.R;


public class LoginActivity extends BaseActivity {
    private String userName, psw, spPsw;
    private EditText et_user_name, et_psw;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_login;
    }

    @Override
    public void initView() {
        TextView mMainTitleTv = $(R.id.titleTv);
        mMainTitleTv.setText(R.string.user_login);
        TextView backTv = $(R.id.backTv);
        backTv.setOnClickListener(i -> this.finish());
        TextView register = $(R.id.registerTv);
        register.setOnClickListener(i -> startActivityForResult(new Intent(this, RegisterActivity.class), 1));
        TextView findPswTV=$(R.id.findPswTv);
        findPswTV.setOnClickListener(i->startActivity(new Intent(this,FindPswActivity.class)));
        TextView loginBtn = $(R.id.btn_login);
        et_user_name = $(R.id.et_user_name);
        et_psw = $(R.id.et_psw);
        register.setOnClickListener(i -> startActivityForResult(new Intent(this, RegisterActivity.class), 1));
        loginBtn.setOnClickListener(i -> {
            userName = et_user_name.getText().toString().trim();
            psw = et_psw.getText().toString().trim();
            String md5 = MD5Utils.md5(psw);
            spPsw = readPsw(userName);
            if (TextUtils.isEmpty(userName)) {
                Toast(getString(R.string.userName));
            } else if (TextUtils.isEmpty(psw)) {
                Toast(getString(R.string.psw));
            } else if (!TextUtils.isEmpty(spPsw) && !md5.equals(spPsw)) {
                Toast(getString(R.string.spPswPsw));
            }else if(spPsw==null){
              Toast("未注册");
            } else if (md5.equals(spPsw)) {
                Toast(getString(R.string.spPsw));
                saveLoginStatus(true, userName);
                startActivity(new Intent(this, MainActivity.class));
                setResult(RESULT_OK, new Intent().putExtra("isLogin", true));
                this.finish();
            }else {
                Toast(getString(R.string.noUser));
            }
        });
    }

    /**
     * 从 SharedPreferences 中根据用户名读取密码
     */
    private String readPsw(String userName) {
        SharedPreferences sp = getSharedPreferences("LoginInfo", MODE_PRIVATE);
            return  sp.getString(userName,"");
    }

    /**
     * 保存登陆状态
     */
    private void saveLoginStatus(boolean status, String userName) {
        SharedPreferences.Editor editor = getSharedPreferences("LoginInfo", MODE_PRIVATE).edit();
        editor.putBoolean("isLogin", status);
        editor.putString("loginUserName", userName);
        editor.apply();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (data != null) {
            String userName = data.getStringExtra("userName");
            if (!TextUtils.isEmpty(userName)) {
                et_user_name.setText(userName);
                et_user_name.setSelection(userName.length());
            }
        }
    }

}

File: app/src/main/java/com/xushengling/myapplication/activity/LoginActivity.kt
package com.xushengling.myapplication.activity

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.text.TextUtils
import com.xushengling.myapplication.R
import com.xushengling.myapplication.base.BaseActivity
import com.xushengling.myapplication.utils.MD5Utils
import kotlinx.android.synthetic.main.activity_login.*
import kotlinx.android.synthetic.main.main_title_bar.*

class LoginActivity : BaseActivity() {

    override fun getLayoutId(): Int {
        return R.layout.activity_login
    }

    override fun initView() {
        titleTv.text=getString(R.string.user_login)
        backTv.setOnClickListener { this.finish()}
        registerTv.setOnClickListener {
            startActivityForResult(Intent(this,
                RegisterActivity::class.java),1)
        }
        findPswTv.setOnClickListener {
            startActivity(Intent(this,FindPswActivity::class.java))
        }
        btn_login.setOnClickListener {
            val userName=et_user_name.text.toString().trim()
            val psw=et_psw.text.toString().trim()
            val md5= MD5Utils.md5(psw)
            val spPsw=getSharedPreferences("loginInfo", Context.MODE_PRIVATE).getString(userName,"")!!
            when{
                TextUtils.isEmpty(userName) -> {
                    toast(getString(R.string.userName))
                }
                TextUtils.isEmpty(psw) -> {
                    toast(getString(R.string.psw))
                }
                md5 == spPsw->{
                    toast(getString(R.string.spPsw))
                    startActivity(Intent(this,
                        MainActivity::class.java))
                    saveLoginStatus(true,userName)
                    setResult(Activity.RESULT_OK,Intent().putExtra("isLogin",true))
                    this.finish()
                }
                !TextUtils.isEmpty(spPsw)&& md5 != spPsw ->{
                    toast(getString(R.string.spPswPsw))
                }
                else ->{
                    toast(getString(R.string.noUser))
                }

            }
        }
    }
    private fun saveLoginStatus(states:Boolean,userName:String){
        val editor=getSharedPreferences("loginInfo", Context.MODE_PRIVATE).edit()
        editor.putBoolean("isLogin",states)
        editor.putString("loginUserName",userName)
        editor.apply()
        editor.commit()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (data!=null){
            val userName=data.getStringExtra("userName")
            if (!TextUtils.isEmpty(userName)){
                et_user_name.setText(userName)
                et_user_name.setSelection(userName!!.length)
            }
        }
    }
}

